packages/tailwindcss/src/index.test.ts

```diff
@@ -948,4 +948,29 @@ describe('Parsing themes values from CSS', () => {
       }"
     `)
   })
+
+  test('utilities that rely on the --default namespace still work when using `@theme reference`', () => {
```

Comment:
Not sure if this test is _really_ needed since this change is reflected in the utilities tests. Maybe better to test this there specifically for the transition utilities with new test cases for providing these two variables both with `@theme` and with `@theme reference`.